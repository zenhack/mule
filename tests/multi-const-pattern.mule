#XFAIL: match not implemented
#
# This should fail to type check, because we have two kinds of
# constants in the same match positions:
fn x. match x with
  | 1 -> x
  | "Hey!" -> x
  | _ -> x
end
