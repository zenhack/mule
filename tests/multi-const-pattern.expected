
Desugared:

	(fn x ((match ((((Text Hey!) x) ((Integer 1) x)) (_ x))) x))
Type error: Mismatched type constructors: int is not a subtype of text
